### PR TITLE
WithOidcAuthority

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #

**Changes proposed in this request**

**Testing**

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 1]
```diff
@@ -268,6 +271,27 @@ private PublicClientApplicationBuilder WithParentFunc(Func<object> parentFunc)
...
         }
 
+        /// <summary>
+        /// Adds a known authority corresponding to a generic OpenIdConnect Identity Provider. 
+        /// MSAL will append ".well-known/openid-configuration" to the authority and retrieve the OIDC 
+        /// metadata from there, to figure out the endpoints.
+        /// See https://openid.net/specs/openid-connect-core-1_0.html#Terminology
+        /// </summary>
+        /// <remarks>
+        /// Experiemental on public clients.
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:281]:
```suggestion
        /// Experimental on public clients.

```

File: src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs [hunk 2]
```diff
@@ -315,19 +316,25 @@ public ConfidentialClientApplicationBuilder WithCacheSynchronization(bool enable
...
         /// MSAL will append ".well-known/openid-configuration" to the authority and retrieve the OIDC 
         /// metadata from there, to figure out the endpoints.
         /// See https://openid.net/specs/openid-connect-core-1_0.html#Terminology
         /// </summary>
-        /// <param name="authorityUri">OpenIdConnect authority</param>
-        /// <returns>The builder to chain the .With methods</returns>
-        /// <remarks>This is an experimental API and only available on Confidential Client flows.</remarks>        
-        public ConfidentialClientApplicationBuilder WithGenericAuthority(string authorityUri)
+        /// <remarks>
+        /// Do not use this method with Entra Id authorities (e.g. https://login.microsfoftonline.com/common).
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:333]:
```suggestion
        /// Do not use this method with Entra Id authorities (e.g. https://login.microsoftonline.com/common).

```

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 3]
```diff
@@ -53,6 +54,8 @@ public static PublicClientApplicationBuilder CreateWithApplicationOptions(Public
                 .WithKerberosTicketClaim(options.KerberosServicePrincipalName, options.TicketContainer);
         }
 
+
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:57]: nit: extra lines

File: tests/devapps/Net5TestApp/Program.cs [hunk 4]
```diff
@@ -1,8 +1,4 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (trwalke) [inline tests/devapps/Net5TestApp/Program.cs:1]: Don't test files need a license too? Or is it only needed for what we distribute?

**Review by trwalke — APPROVED** [review]

File: tests/devapps/Net5TestApp/Program.cs (hunk 4 above)
Comment (pmaytak) [reply tests/devapps/Net5TestApp/Program.cs:1]: Afaik everything.

**Pull request merged**